src/main/java/edu/harvard/iq/dataverse/api/Datasets.java

```diff
@@ -1161,6 +1231,17 @@ public Response addFileToDataset(@PathParam("id") String idSupplied,
             return wr.getResponse();           
         }
         
+        //------------------------------------
+        // (2a) Make sure dataset does not have package file
+        //
+        // --------------------------------------
+        
+        if (dataset.getEditVersion().isHasPackageFile()){
```

Comment:
isHas?

Reply:
That bit of code is from dual mode. I closed this pr because the comparison isn't accurate.